src/commands/registry.rs

```diff
@@ -90,3 +94,65 @@ impl Pull {
         Ok(())
     }
 }
+
+#[derive(Parser, Debug)]
+pub struct Login {
+    /// Username for the container registry
+    #[clap(long = "username", short = 'u')]
+    pub username: Option<String>,
+
+    /// Password for the container registry
+    #[clap(long = "password", short = 'p')]
+    pub password: Option<String>,
+
+    /// Take the password from stdin
+    #[clap(
+        long = "password-stdin",
+        takes_value = false,
+        conflicts_with = "password"
+    )]
+    pub password_stdin: bool,
+
+    #[clap()]
+    pub server: String,
+}
+
+impl Login {
+    pub async fn run(self) -> Result<()> {
+        let username = match self.username {
+            Some(u) => u,
+            None => {
+                println!("Username: ");
+                let mut u = String::new();
+                std::io::stdin().read_line(&mut u)?;
+
+                u
+            }
+        };
+
+        // If the --password-stdin flag is passed, read the password from standard input.
+        // Otherwise, if the --password flag was passed with a value, use that value. Finally, if
+        // neither was passed, prompt the user to input the password.
+        let password = if self.password_stdin {
+            let mut buf = String::new();
+            let mut stdin = std::io::stdin().lock();
+            stdin.read_to_string(&mut buf)?;
+            buf
+        } else {
+            match self.password {
+                Some(p) => p,
+                None => rpassword::prompt_password("Password:")?,
+            }
+        };
+
+        Client::login(&self.server, &username, &password)
+            .await
+            .context("cannot login to the registry")?;
```

Comment:
for consistency with the help text and because pedantry

```suggestion
            .context("cannot log in to the registry")?;
```